File: LeetCode/484-find-permutation/ex_484.py
class Solution(object):
    def findPermutation(self, s):
        """
        :type s: str
        :rtype: List[int]
        """
        if not s:   return []
        res = []
        i = 1
        for c in s:
            if c == 'I':
                res.extend(range(i, len(res), -1))
            i += 1
        res.extend(range(i, len(res), -1))
        return res

File: LeetCode/484-find-permutation/ex_484.cpp
class Solution {
public:
    vector<int> findPermutation(string s) {
        vector<int> res;
        int i = 1;
        for (char &c : s) {
            if (c == 'I') 
                for (int j = i, length = res.size(); j != length; --j)
                    res.push_back(j);
            ++i;
        }
        for (int j = i, length = res.size(); j != length; --j)
            res.push_back(j);
        return res;
    }
};